### PR TITLE
RUM-4151 [SR] Support background color for Tab Bars

### PR DESCRIPTION
**Pull request by mariedm**
### What and why?

Session Replay currently does not support custom background colors for tab bars, defaulting to white or black based on the user's trait collection.

### How?

This PR:
- Modifies the SR Tab Bar Recorder to support custom background colors.
- Adds new snapshot tests for both static and embedded tab bars.

This work is similar to a previous PR that added support for custom background colors on navigation bars. For more context, see [PR #1864](https://github.com/DataDog/dd-sdk-ios/pull/1864).

## Notes on deselected items’ Tint Color

TLDR: if no `unselectedTintColor` is set, icons will be rendered in white.

- Both selected and unselected tint colors for tab bar items can be set using `tintColor` and `unselectedItemTintColor`.
- If `unselectedItemTintColor` is not set, the default is gray, though it appears white in the view debugger and is not accessible via code.
- An item's icon is recorded using the `UIImageViewRecorder`. The correct tint color for a selected item is obtained through the `tintColorProvider`.
- It is not possible to detect selection status at the `image` or `imageView` level within the `UIImageViewRecorder`.
    - `image.isTinted` always returns `true` if a `tintColor` is set, regardless of selection status.
    - Without access to the unselected tint color, we cannot determine the rendered color for unselected items.
    - Comparing the recorded image with the `withTintColor(_:)` effect applied to it does not work.
    - Accessing the Tab Bar and Tab Bar button from the image's superviews is possible, but the button is of type `UITabBarButton`, a private API, preventing state checks to determine selection status and appropriate tint color.
## Snapshot tests screenshots

**Static Tab Bar**

testTabBars()-allow-privacy

**Embedded Tab Bar**

testTabBars()-embeddedtabbar-allow-privacy

### Review checklist
- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes

### Custom CI job configuration (optional)
- [ ] Run unit tests for Core, RUM, Trace, Logs, CR and WVT
- [ ] Run unit tests for Session Replay
- [ ] Run integration tests
- [ ] Run smoke tests
- [ ] Run tests for `tools/`

### CODE REVIEW

File: DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UITabBarRecorder.swift
Comment (ncreated) [inline DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UITabBarRecorder.swift:28]: 🎯👌

**Review by ncreated — APPROVED** [review]
Looks great 🎯. I appreciate the level of details in PR description and snapshot tests - all clear 🚀! 

Minor comment on conventions: we usually mention the JIRA ref (RUM-4151 in this case) as part of every commit (prefix) and in the PR title. This way we establish automatic link between GH and JIRA, which helps tracking decisions later. For this PR, let's mention it in the title at least 🙂.

File: DatadogSessionReplay/SRSnapshotTests/SRFixtures/Sources/SRFixtures/ViewControllers/TabbarControllers.swift
Comment (maxep) [inline DatadogSessionReplay/SRSnapshotTests/SRFixtures/Sources/SRFixtures/ViewControllers/TabbarControllers.swift:17]: /nit
```suggestion
            tabbar.selectedItem = tabbar.items?.first
```

**Review by maciejburda — COMMENTED** [review]
Nice!

I would only suggest defaulting the icon image tint to gray whenever we deal with `nil`.

Also - we probably can drop the bottom border. Seems like native component has only a gray line on top.

**Review by maxep — DISMISSED** [review]
Looks great! nice work on the snapshot tests 👏

**Comment by mariedm** [comment]
> Minor comment on conventions: we usually mention the JIRA ref (RUM-4151 in this case) as part of every commit (prefix) and in the PR title. This way we establish automatic link between GH and JIRA, which helps tracking decisions later. For this PR, let's mention it in the title at least 🙂.

Got it! I will follow this pattern next time. In the meantime, I added it to the PR title.

**Comment by mariedm** [comment]
> I would only suggest defaulting the icon image tint to gray whenever we deal with `unselectedItemTintColor == nil`.
> 
> Also - we probably can drop the bottom border. Seems like native component has only a gray line on top.

@maciejburda I can remove the bottom border color. 👍 However, how do you suggest we default the icon to gray when `unselectedItemTintColor == nil`? We don't have access to this property within the `UIImageViewRecorder`, and we can't determine if the tab bar item it belongs to is selected or not. Therefore, we don't know whether to apply the `tintColor` or the `unselectedItemTintColor`.

**Comment by mariedm** [comment]
@maciejburda Actually, the current implementation in `UITabBarRecorder` will also remove the top border, which might not be what we want?

testTabBars()-mask-privacy

testTabBars()-embeddedtabbar-allow-privacy

**Comment by mariedm** [comment]
@maciejburda as discussed in the Daily today, I can merge this PR once I have the required approvals to unblock @ncreated's work. I created a follow-up [ticket](https://datadoghq.atlassian.net/browse/RUM-4883) from your comment.

**Review by maciejburda — APPROVED** [review]
LGTM!

As discussed - we can try to tackle white tint images in a separate PR

**Pull request merged**
